### ISSUE TITLE
[doxygen] EXTRACT_ALL and EXTRACT_PRIVATE

### ISSUE THREAD

**Issue by jgvictores** [roboticslab-uc3m/project-generator#22]
https://stackoverflow.com/questions/9484879/graphviz-doxygen-to-generate-uml-class-diagrams/9488742#9488742

```
EXTRACT_ALL            = YES
HAVE_DOT               = YES
UML_LOOK               = YES
```

**Comment by jgvictores** [comment]
Added EXTRACT_PRIVATE to description too.

**Comment by jgvictores** [comment]
https://github.com/asrob-uc3m/robotDevastation/commit/af27e91e5b1db9010c9f2221ae2c4fcc753df77e
and
https://github.com/asrob-uc3m/robotDevastation/commit/6ad771987836ce0f9140409e169890883432479b

**Comment by PeterBowman** [comment]
Currently enforcing EXTRACT_PRIVATE via CMake-to-Doxyfile:

https://github.com/roboticslab-uc3m/project-generator/blob/6390b0ea23744ce5941a782ec2a615fccbe7b7be/doc/CMakeLists.txt#L20

**Comment by PeterBowman** [comment]
[`EXTRACT_ALL`](https://www.doxygen.nl/manual/config.html#cfg_extract_all):

> If the `EXTRACT_ALL` tag is set to `YES`, doxygen will assume all entities in documentation are documented, even if no documentation was available. Private class members and static file members will be hidden unless the `EXTRACT_PRIVATE` respectively `EXTRACT_STATIC` tags are set to `YES`.
>
> **Note**
> This will also disable the warnings about undocumented members that are normally produced when `WARNINGS` is set to `YES`.

[`EXTRACT_PRIVATE`](https://www.doxygen.nl/manual/config.html#cfg_extract_private):

> If the `EXTRACT_PRIVATE` tag is set to `YES`, all private members of a class will be included in the documentation.

Changes referenced at https://github.com/roboticslab-uc3m/project-generator/issues/22#issuecomment-404967117 seem unconnected to the title, which explains nothing. I could assume the point of this issue was having UML graphs in doxy, but I'm not psychic. For now, I'd be happy with the CMake handling of Doxyfiles, which was added at https://github.com/roboticslab-uc3m/project-generator/commit/3e79392fe08bcd09209ac798034523e94d1404fc and incidentally solves the `EXTRACT_PRIVATE` thing, while I still have no clue about what we need `EXTRACT_ALL` for.

Let's close this and discuss UML stuff elsewhere.

**Comment by jgvictores** [comment]
Just for the historic: The issue objective was not related to UML at all. https://github.com/asrob-uc3m/robotDevastation/commit/6ad771987836ce0f9140409e169890883432479b was referenced because `EXTRACT_PRIVATE` was set to `YES`.

There are two aspects here:

- Where these parameters are activated: The current situation of CMake handling Doxyfiles is great, thank you!
- Which parameters should be activated: When I opened the issue, I thought it would be good for Doxygen to show all file members even if undocumented. Currently, I have no clear preference.

To sum up, it's fine to close the issue. Cheers!
